content/livesync/models.textile

```diff
@@ -0,0 +1,354 @@
+
+When a message is received on the channel, the merge function is called to merge that new message into the existing Model state. The merge function is called with the existing Model state and the newly received channel message, and it should return the updated Model state (with the new message merged in).
+
+It's important that the merge function is pure and deterministic, and not depending on any external state. This is because the merge will often be called multiple times and in different orders (for example when replaying events after rewinding to a sequenceId, or when rebasing optimistic events on top of confirmed events).
+
+The following example is a merge function, invoked for all events received on the channel name specified in the model:
+
+```[javascript]
+function merge(state: Post, event: OptimisticEvent | ConfirmedEvent) {
+  if (event.name === 'addComment') {
+     return {
+    ...state,
+    comments: state.comments.concat([event.data]),
+  };
+   }
+ // handle other event types
+}
+```
+
+The merge function needs to be registered when a model is "instantiated:":#create
+
+```[javascript]
+const model = modelsClient.models.get({
+  channelName: 'posts:123',
+  sync,
+  merge,
+});
+```
+
+The event passed to the merge function can be either confirmed or optimistic:
+
+- Confirmed := change events received from your backend. They describe the result of a change to the data which has been committed to your database.
+- "Optimistic":#optimistic := events that describe mutations that have happened locally, but have not yet been confirmed by your backend.
+
+The SDK might call the merge function multiple times with the same state and event; to rebase optimistic events onto the newly confirmed state. It's essential that the merge function, when executed with identical inputs, consistently produces the same output. The merge function should be pure and deterministic, not depending on any external state.
+
+h3(#event-buffer). Event buffer
+
+Change events from your backend are delivered to the SDK as messages over the "channels":/channels. By default, messages are processed by the SDK in the order in which they are received.
+
+The SDK supports a short time buffering for change events, facilitating short-term reordering and de-duplication. By default, this feature is not enabled. To activate it, set the @eventBufferOptions.bufferMS@ in the "@ClientOption@":#client-options to a non-zero value.
+
+When using the event buffer, change events are de-duplicated and ordered according to their "@message.id@":/api/realtime-sdk/messages#id, which corresponds to the change event's "@sequenceId@":#sequence.
+
+By default, the events in the buffer are ordered numerically if the @message.id@ can be coerced to a number. Otherwise, events will be ordered lexicographically by their @message.id@. You can specify a custom ordering based on any part of the message via the @eventBufferOptions.eventOrderer@ "client option":#client-options.
```

Comment:
```suggestion
By default, the events in the buffer are ordered numerically if the @message.id@ can be coerced to a number. Otherwise, events will be ordered lexicographically by their @message.id@. You can specify a custom ordering based on any part of the message via the @eventBufferOptions.eventOrderer@ "@ClientOption@":#client-options.
```